src/main/scala/fs2/aws/kinesis/kpl.scala

```diff
@@ -0,0 +1,44 @@
+package fs2
+package aws
+package kinesis
+
+import com.google.common.util.concurrent.{ListenableFuture, FutureCallback, Futures}
+import cats.effect.{Effect, Concurrent}
+import fs2.aws.internal.Internal._
+import com.amazonaws.services.kinesis.producer.UserRecordResult
+import scala.concurrent.ExecutionContext
+
+
+package object publisher {
+  def writeToKinesis[F[_]](
+    streamName: String,
+    partitionKey: String,
+    producer: KinesisProducerClient[F] = new KinesisProducerClient[F] {}
+  )(implicit F: Effect[F], ec: ExecutionContext, concurrent: Concurrent[F]): fs2.Pipe[F, Byte, Either[Throwable, UserRecordResult]] = {
```

Comment:
should we have an overloaded method for writeToKinesis that is just a fs2.Sink[F, Byte], it could discard the UserRecordResults and give a simpler API ?

Reply:
that's reasonable. I actually initially started with a Sink but then opted for a Pipe so we can return the results. I'll add a Sink as well

Reply:
scalac doesn't allow for overloaded methods with default params, so I'll create `writeToKinesis_`, inspired by `Stream.eval_` (https://oss.sonatype.org/service/local/repositories/releases/archive/co/fs2/fs2-core_2.12/1.0.0/fs2-core_2.12-1.0.0-javadoc.jar/!/fs2/Stream$.html#eval_[F[_],A](fa:F[A]):fs2.Stream[F,fs2.INothing])

Reply:
but see, eval_ reuses eval https://github.com/functional-streams-for-scala/fs2/blob/7146253402a32181ebbece60a9745de0756bb3c1/core/shared/src/main/scala/fs2/Stream.scala#L2672

Where you're actually copy pasting the logic here :/

Reply:
whoops - good call out